### PR TITLE
Improve mime type filtering

### PR DESCRIPTION
**Pull request by roborourke**
### Description of the Change

As noted in issue #1604 the image block sends a mime type value of `image` through as an array which goes against the convention this part of the query filter relied on. This update makes the filter more agnostic, working with a single string (as `wp_post_mime_type_where()` does) or an array, plus it will then separate the values into full terms and prefixes.

### Benefits

This has the benefit of supporting almost any combination of values given in the `post_mime_type` argument and avoids the use of a `regexp` query which can be slower to run.

This fixes issue #1604

### Verification Process

Searching media in the media modal opened from an image block in the block editor now returns results.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Applicable Issues

- #1604 

### Changelog Entry

- Changed `post_mime_type` filtering to support more edge cases

### CODE REVIEW

**Comment by roborourke** [comment]
I'll look into updating the tests too, just wanted to get this up.

**Comment by roborourke** [comment]
Ah seems the test suite is a bit broken right now.

File: includes/classes/Indexable/Post/Post.php [hunk 1]
```diff
@@ -907,25 +907,61 @@ function( $tax_query ) use ( $args ) {
...
+		 * or an array of mime types can be passed 
 		 *
 		 * @since 2.3
 		 */
 		if ( ! empty( $args['post_mime_type'] ) ) {
-			if ( is_array( $args['post_mime_type'] ) ) {
-				$filter['bool']['must'][] = array(
+			if ( is_string( $args['post_mime_type'] ) ) {
+				$mime_types = array_map( 'trim', explode( ',', $args['post_mime_type'] ) );
+			} else {
```
Comment (tlovett1) [inline includes/classes/Indexable/Post/Post.php:918]: This doesn't make total sense to me. I think it would be better to check if it's a string, then check if it's an array. If it's anything else, ignore the parameter.

File: includes/classes/Indexable/Post/Post.php [hunk 2]
```diff
@@ -907,25 +907,61 @@ function( $tax_query ) use ( $args ) {
...
 				);
+			}
 
-				$use_filters = true;
-			} elseif ( is_string( $args['post_mime_type'] ) ) {
+			if ( ! empty( $prefixes ) ) {
+				foreach ( $prefixes as $prefix ) {
+					$mime_type_filter[] = array(
+						'prefix' => array(
+							'post_mime_type' => $prefix,
```
Comment (tlovett1) [inline includes/classes/Indexable/Post/Post.php:954]: Is mime type supposed to support wildcards?

**Review by tlovett1 — CHANGES_REQUESTED** [review]

File: includes/classes/Indexable/Post/Post.php (hunk 2 above)
Comment (roborourke) [reply includes/classes/Indexable/Post/Post.php:954]: Yeah but only in a limited way - this function is used to build the SQL in WP core https://developer.wordpress.org/reference/functions/wp_post_mime_type_where/

You can query by subgroup eg. `image` or by fully qualified mime types like `image/jpeg`

File: includes/classes/Indexable/Post/Post.php (hunk 1 above)
Comment (roborourke) [reply includes/classes/Indexable/Post/Post.php:918]: Yep, good shout 👍

**Comment by felipeelia** [comment]
Closing this as we've merged #2222 to fix the same issue.
